### PR TITLE
Add CTA experiments

### PR DESCRIPTION
**Pull request by subsymbolic**
Task/Issue URL: https://app.asana.com/0/361428290920652/1158110040574203
Tech Design URL: N/A

**Description**:
Adds CTA retention experiments to help up determine effectiveness of CTAa

**Steps to test this PR**:

**Check CTAs are shown for control:**
1. Open VariantManager.kt and comment out all variants other than `mq` control
1. Run a fresh installation of the app and launch it to the welcome screen
1. Ensure the `m_odb_v` pixel has NOT fired yet
1. Press `Continue`
1. Ensure the the `Make DuckDuckGo Your Default` onboarding is shown and the `m_odb_v` pixel now fires
1. Select `Maybe later` to get to the home screen
1. Ensure the widget CTA shows

**Check that only widget CTA is shown on high end device and triggers correct pixels when default browser CTA feature is suppressed:**
1. Open VariantManager.kt and comment out all variants other than `mr`
1. Launch the app on an API 26 above emulator or device that supports widget pinning
1. Run a fresh installation of the app and launch it to the welcome screen, press `Start Browsing`
1. Ensure the widget CTA shows and the `m_wc_s` pixel fires
1. Press `Add Widget` and ensure the `m_wc_l` pixel fires
1. Press `Add Automatically` and ensure the CTA is gone
--
7. Reset the app and run through the previous steps however this time select `Dismiss` at step 4
1. Ensure the `m_wc_d` pixel fires 
1. Ensure the Add widget CTA is now gone

**Check that only widget CTA is shown on low end device and triggers correct pixels when default browser CTA feature is suppressed:**
1. Open VariantManager.kt and comment out all variants other than `mr`
1. Launch the app on an API 22-25 emulator or device
1. Run a fresh installation of the app and launch it to the welcome screen, press `Start Browsing`
1. Ensure the widget CTA shows and the `m_wlc_s` pixel fires
1. Press `Show Me` and ensure the `m_wlc_l` pixel fires
1. Press `Go to home screen` and add the widget to the home screen
1. Relaunch the app, close the info screen and ensure the CTA is gone
--
7. Reset the app and run through previous steps 1-4 however this time select `Dismiss` at step 4
1. Ensure the `m_wlc_d` pixel fires 
1. Ensure the CTA is now gone

**Check only default browser CTA shown when widget CTA feature suppressed:**
1. Open VariantManager.kt and comment out all variants other than `ms`
1. Run a fresh installation of the app and launch it to the welcome screen
1. Ensure the `m_odb_v` pixel has NOT fired yet
1. Press `Continue`
1. Ensure the the `Make DuckDuckGo your default` onboarding is shown and the `m_odb_v` pixel now fires
1. Select `Maybe later` to get to the home screen
1. Ensure the widget CTA is not shown

**Check that no CTA shown when they are suppressed:**
1. Open VariantManager.kt and comment out all variants other than `mt`
1. Run a fresh installation of the app and launch it to the welcome screen and press  press `Start Browsing`
1. Note that no CTAs were displayed

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: app/src/main/java/com/duckduckgo/app/onboarding/ui/page/DefaultBrowserPage.kt [hunk 1]
```diff
@@ -63,6 +63,13 @@ class DefaultBrowserPage : OnboardingPageFragment() {
         super.onAttach(context)
     }
 
+    override fun setUserVisibleHint(isVisibleToUser: Boolean) {
```
Comment (marcosholgado) [inline app/src/main/java/com/duckduckgo/app/onboarding/ui/page/DefaultBrowserPage.kt]: We are still not using the latest version of fragment-ktx but this method has been deprecated in 1.1.0 (https://developer.android.com/jetpack/androidx/releases/fragment). Probably not something for this PR but maybe create a LHF task?

File: app/src/androidTest/java/com/duckduckgo/app/cta/ui/CtaViewModelTest.kt [hunk 2]
```diff
@@ -219,24 +215,24 @@ class CtaViewModelTest {
     }
 
     @Test
-    fun whenRefreshCtaOnNewTabThenReturnDaxIntroCta() = runBlockingTest {
-        setConceptTestVariant()
+    fun whenRefreshCtaOnNewTabAndConceptTestFeatureActiveThenReturnDaxIntroCta() = runBlockingTest {
```
Comment (cmonfortep) [inline app/src/androidTest/java/com/duckduckgo/app/cta/ui/CtaViewModelTest.kt]: Hey @subsymbolic, https://github.com/duckduckgo/Android/pull/673 gave you some conflicts here. Happy to help you in solving this later today.

I will still put some context here:
* we renamed/mapped `isNewTab` to `isBrowserShowing` to show a better intention inside our `ctaViewModel.refreshCta`.
`isNewTab = false` equals `isBrowserShowing = true`
`isNewTab = true` equals `isBrowserShowing = false`
* testcases names have been affected by the change too:
`whenRefreshCtaOnNewTab` -> `whenRefreshCtaOnHomeTab`
`whenRefreshCtaOnExistingTab` -> `whenRefreshCtaWhileBrowsing`

Comment (subsymbolic) [reply app/src/androidTest/java/com/duckduckgo/app/cta/ui/CtaViewModelTest.kt]: Thanks!

File: app/src/main/java/com/duckduckgo/app/onboarding/ui/page/DefaultBrowserPage.kt (hunk 1 above)
Comment (subsymbolic) [reply app/src/main/java/com/duckduckgo/app/onboarding/ui/page/DefaultBrowserPage.kt]: Good idea, internal task created here https://app.asana.com/0/414730916066338/1158589410650834

**Comment by subsymbolic** [comment]
Thanks for the review @marcosholgado

**Pull request merged**